src/output.rs

```diff
@@ -28,7 +28,16 @@ impl TestResult {
     pub fn fail(name: String, message: Option<String>) -> TestResult {
         TestResult {
             name: format_test_name(name),
-            message,
+            message: message.map(|m| {
+                // This note is attached to the error message of only one test case that fails,
+                // but not always the same one. To avoid CI failing unnecessarily, this note
+                // is stripped from all messages.
+                // It's also not useful to students reading the output of the test runner,
+                // as they can't set this environment variable in the test runner themselves.
+                m.trim_end_matches(
+                    "note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace\n"
+                ).to_owned()
+            }),
```

Comment:
@ErikSchierboom I was able to reproduce, but it only fails *sometimes*. I ran `bin/run-tests-in-docker.sh` in a loop, the first two times succeeded with the existing `expected_results.json`, the third one failed. Since the output of `cargo test` seems to be non-deterministic, I think it won't be enough to change `expected_results.json`.

I don't yet know what determines which of the test cases gets this additional note:
```
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace\n
```
I can only speculate it's the first one that fails. And if the tests are run in parallel, this may be a different one between different runs.

With these changes, I was able to run the tests in 50 loop iterations successfully, so I believe this does fix the flakiness. I can't think of a downside to simply stripping this message from the output of `cargo test`, let me know if you can think of one.

Reply:
Completely agree. The backtrace mention is actually unhelpful as people can't enable it. Good call!